addons/stock_landed_costs/models/account_move.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    landed_costs_ids = fields.One2many('stock.landed.cost', 'vendor_bill_id', string='Vendor Bills')
+    landed_costs_visible = fields.Boolean(compute='_compute_landed_costs_visible')
+
+    @api.depends('line_ids', 'line_ids.is_landed_costs_line')
+    def _compute_landed_costs_visible(self):
+        for account_move in self:
+            if account_move.landed_costs_ids:
+                account_move.landed_costs_visible = False
+            else:
+                account_move.landed_costs_visible = any(line.is_landed_costs_line for line in account_move.line_ids)
+
+    def button_create_landed_costs(self):
+        """Create a `stock.landed.costs` record associated to the account move of `self`, each
+        `stock.landed.costs` lines mirroring the current `account.move.line` of self.
+        """
+        self.ensure_one()
+        landed_costs_lines = self.line_ids.filtered(lambda line: line.is_landed_costs_line)
+
+        landed_costs = self.env['stock.landed.cost'].create({
+            'vendor_bill_id': self.id,
+            'cost_lines': [(0, 0, {
+                'product_id': l.product_id.id,
+                'name': l.product_id.name,
+                'account_id': l.product_id.product_tmpl_id.get_product_accounts()['stock_input'].id,
+                'price_unit': l.price_subtotal,
+                'split_method': 'equal',
+            }) for l in landed_costs_lines],
+        })
+        action = self.env.ref('stock_landed_costs.action_stock_landed_cost').read()[0]
+        return dict(action, view_mode='form', res_id=landed_costs.id, views=[(False, 'form')])
+
+    def action_view_landed_costs(self):
+        self.ensure_one()
+        action = self.env.ref('stock_landed_costs.action_stock_landed_cost').read()[0]
+        domain = [('id', 'in', self.landed_costs_ids.ids)]
+        context = dict(self.env.context, default_vendor_bill_id=self.id)
+        views = [(self.env.ref('stock_landed_costs.view_stock_landed_cost_tree2').id, 'tree'), (False, 'form'), (False, 'kanban')]
+        return dict(action, domain=domain, context=context, views=views)
+
+
+class AccountMoveLine(models.Model):
+    _inherit = 'account.move.line'
+
+    product_type = fields.Selection(related='product_id.type', readonly=True)
+    is_landed_costs_line = fields.Boolean()
+
+    @api.onchange('is_landed_costs_line')
+    def _onchange_is_landed_costs_line(self):
```

Comment:
I would keep only
```python
if self.is_landed_costs_line and self.move_id.company_id.anglo_saxon_accounting:
                self.account_id = accounts['stock_input']
```
and set a domain on is_landed_cost_line (produc_type, '=', 'service')

Reply:
there's a domain in the view
> attrs="{'readonly': [('product_type', '!=', 'service')]}"